controllers/profile.js

```diff
@@ -19,21 +19,39 @@ class Profiles {
     const profile = await Profile.findOne({
       where: { userId: req.decoded.id }
     });
-    const userProfile = await profile
+
+    const {
+      firstName,
+      lastName,
+      bio,
+      imageUrl,
+      interests = []
+    } = req.body;
+
+    let normalizedInterests;
+
+    normalizedInterests = interests;
+    if (profile.interests) {
+      normalizedInterests = [
+        ...new Set(profile.interests.concat(interests))
+      ];
+    }
+
+    const responseData = await profile
       .update({
-        firstName: req.body.firstName || profile.firstName,
-        lastName: req.body.lastName || profile.lastName,
-        bio: req.body.bio || profile.bio,
-        interests: req.body.interests || profile.interests,
-        imageUrl: req.body.imageUrl || profile.imageUrl
+        firstName: firstName || profile.firstName,
+        lastName: lastName || profile.lastName,
+        bio: bio || profile.bio,
+        interests: normalizedInterests || profile.interests,
```

Comment:
We won't need the second part of this branch anymore. Already handled on line 33.